lib/tests/formulae_detect.rb

```diff
@@ -73,8 +73,8 @@ def detect_formulae!(args:)
             diff_end_sha1 = github_sha
           # Use GitHub Actions variables for merge group jobs.
           elsif ENV.fetch("GITHUB_EVENT_NAME", nil) == "merge_group"
-            origin_ref = "origin/#{github_ref.gsub(%r{^refs/heads/}, "")}"
             diff_start_sha1 = rev_parse(origin_ref)
+            origin_ref = "origin/#{github_ref.gsub(%r{^refs/heads/}, "")}"
             diff_end_sha1 = github_sha
           # Use GitHub Actions variables for branch jobs.
```

Comment:
Just for future reference: this is the bit that should be fixed for the `push` job to work.